clang/lib/Parse/ParseDecl.cpp

```diff
@@ -4504,6 +4504,11 @@ void Parser::ParseDeclarationSpecifiers(
       isInvalid = DS.setFunctionSpecNoreturn(Loc, PrevSpec, DiagID);
       break;
 
+    case tok::kw__Export:
+      // We're done with the declaration-specifiers.
+      goto DoneWithDeclSpec;
```

Comment:
Does this mean `_Export` has to be last?   Not a parser guy here, but this one doesn't feel right to me?  @aaronballman can you double check here?

Reply:
`_Export` is not really a declaration specifier (at least in the traditional implementation).
It is more like a C++11 attribute except that it lives on the left of a _declarator-id_ instead of on the right.
In some realizations, the (less clean) conceptual model is that it is a declarator like `*` is, except it binds more strongly than `(<function parameters>)` or `[<array bound>]`.

@perry-ca, I don't see any test in this PR for `_Export` in positions like:
```cpp
int (*_Export x)(void) = 0;
```

Is the above accepted with this PR? Which conceptual model is being implemented here?